File: examination-management/src/main/java/com/nextgen/erp/examination_management/entity/ExamSchedule.java
package com.nextgen.erp.examination_management.entity;

import com.nextgen.erp.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

@Entity
@Table(
        name = "exam_schedules",
        uniqueConstraints = {
                @UniqueConstraint(
                        columnNames = {
                                "exam_id",
                                "subject_id",
                                "section_id"
                        }
                )
        }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ExamSchedule extends BaseEntity {

    @Column(name = "exam_id", nullable = false)
    private UUID examId;

    @Column(name = "subject_id", nullable = false)
    private UUID subjectId;

    @Column(name = "teacher_id", nullable = false)
    private UUID teacherId;

    @Column(name = "class_id", nullable = false)
    private UUID classId;

    @Column(name = "section_id", nullable = false)
    private UUID sectionId;

    @Column(name = "exam_date", nullable = false)
    private LocalDate examDate;

    @Column(name = "start_time", nullable = false)
    private LocalTime startTime;

    @Column(name = "end_time", nullable = false)
    private LocalTime endTime;

    @Column(name = "room_number", length = 50)
    private String roomNumber;

    @Column(name = "max_marks", nullable = false)
    private Integer maxMarks;
}

File: frontend/src/types/examSchedule.ts
export interface ExamSchedule {
  id: string;
  examId: string;
  subjectId: string;
  teacherId: string;
  classId: string;
  sectionId: string;
  examDate: string;
  startTime: string;
  endTime: string;
  roomNumber?: string;
  maxMarks: number;
}

export interface ExamScheduleRequest {
  examId: string;
  subjectId: string;
  teacherId: string;
  classId: string;
  sectionId: string;
  examDate: string;
  startTime: string;
  endTime: string;
  roomNumber?: string;
  maxMarks: number;
}
